### DISCUSSION TITLE
How to use grpc-web in tauri?

### DISCUSSION THREAD

**Discussion by NicFTK** [tauri-apps/tauri#10444] [Q&A]
I would like to use grpc-web in Tauri, but in a Tauri application, the webview can only connect to the network through its own HTTP module. Is there any way to use grpc-web in Tauri?

**Comment by NicFTK** [comment]
I had Searched on GitHub and Stack Overflow.
And tried running grpc-web directly in Tauri, but encountered errors.
{
    "message": "Http response at 400 or 500 level, http status code: 0",
    "stack": "Error: Http response at 400 or 500 level, http status code: 0\n    at new E (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:23:64)\n    at Y (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:59:40)\n    at rc.eval (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:57:196)\n    at Kb (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:35:182)\n    at O (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:34:614)\n    at zc (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:45:411)\n    at Bc (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:48:228)\n    at n.W (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:46:252)\n    at n.R (webpack://nico_service/./node_modules/.pnpm/grpc-web@1.5.0/node_modules/grpc-web/index.js?:46:231)",
    "code": 2,
    "metadata": {}
}

**Comment by FabianLars** [comment]
As far as i am aware, grpc-web should "just work" in Tauri. There are a few issues like this one for example https://github.com/tauri-apps/tauri/issues/9288 (macos) but that also implies that the it generally works.

Which OS / OS version are we talking about here?

Are you sure it's not a server side issue?

**Comment by NicFTK** [comment]
Thank you for your response. I spent quite a bit of time troubleshooting the issue, which made me doubt whether grpc-web could be used in Tauri. After rechecking my code, I found that it was indeed a configuration error. Now, it is running correctly in my Tauri app.
